test/helpers.tsx

```diff
@@ -1,24 +1,102 @@
-import { ReactElement } from "react";
+import { ComponentProps, ComponentType, ReactElement, useState } from "react";
 import { StylesProvider, jssPreset } from "@material-ui/core/styles";
 import { create } from "jss";
+import { MuiPickersUtilsProvider } from "@material-ui/pickers";
+import DateFnsUtils from "@date-io/date-fns";
 
 export function render(element: ReactElement) {
-  let insertion = bigtestGlobals.document.createComment("mui-jss-insertion");
-  let insertionPoint = bigtestGlobals.document.head.insertBefore(insertion, bigtestGlobals.document.head.firstChild);
-  const jss = create({
-    ...jssPreset(),
-    // Define a custom insertion point that JSS will look for when injecting the styles into the DOM.
-    insertionPoint,
-  });
-
-  return rtlRender(
-    <StylesProvider jss={jss} injectFirst>
-      {element}
-    </StylesProvider>,
-    {
-      container: bigtestGlobals.document.body,
-    }
-  );
+  return {
+    description: "render",
+    action: () => {
+      let insertion = document.createComment("mui-jss-insertion");
+      let insertionPoint = document.head.insertBefore(insertion, document.head.firstChild);
+      const jss = create({
+        ...jssPreset(),
+        // Define a custom insertion point that JSS will look for when injecting the styles into the DOM.
+        insertionPoint,
+      });
+
+      rtlRender(
+        <StylesProvider jss={jss} injectFirst>
+          {element}
+        </StylesProvider>,
+        {
+          container: document.body,
+        }
+      );
+    },
+  };
+}
+
+type HTMLTypes<T> = T extends `HTML${infer C}Element` ? C : never;
+
+type HTMLElementTypes = HTMLTypes<keyof typeof window>;
+
+export function isHTMLElement<T extends HTMLElementTypes = "">(
+  element: unknown | null | undefined,
+  type: T = "" as T
+): element is InstanceType<typeof window[`HTML${T}Element`]> {
+  const { defaultView } = bigtestGlobals.document;
+  const Constructor = (defaultView as any)?.[`HTML${type}Element`];
+  return typeof Constructor == "function" && element instanceof Constructor;
+}
+
+export interface WrapperProps<CT extends ComponentType<any>> {
+  getProps?: Partial<ComponentProps<CT>> | ((props?: Partial<ComponentProps<CT>>) => Partial<ComponentProps<CT>>);
+  props?: Partial<ComponentProps<CT>>;
+  children: (props?: Partial<ComponentProps<CT>>) => ReactElement<ComponentProps<CT>, CT>;
+}
+
+export function getComponentRenderer<CT extends ComponentType<any>>(
+  Component: CT,
+  defaultProps: Partial<ComponentProps<CT>> = {},
+  Wrapper: ComponentType<WrapperProps<CT>> = ({ getProps, children }) =>
+    children(typeof getProps == "function" ? getProps() : getProps)
+) {
+  return (getProps?: WrapperProps<CT>["getProps"]) =>
+    render(
+      <Wrapper getProps={getProps} props={typeof getProps == "function" ? getProps() : getProps}>
+        {(props) => <Component {...({ ...defaultProps, ...props } as ComponentProps<CT>)} />}
+      </Wrapper>
+    );
+}
```

Comment:
1. Let's call this `createRenderStep` so that it's very specific (and maybe even have the return type be a `StepImplementation`
2. We have a lot of information here that we can use to make the step description very helpful in debugging and we should use it. For example, we have the props. Maybe we should make this take a string so we can use it here to. Example:

```ts
const renderCalendar = createRendererStep("Calendar", CalendarComponent);
```
That way, if we get a failure, then in the step trace, we will see it as:

"render Calendar({ max: 2014, min: 2000})"

or something like that.

Reply:
1. What is the `StepImplementation`? Can't find that type
2. I can use `Component.displayName` here and make the string argument optional.

Reply:
1. Would be enough it?
```ts
interface StepImplementation {
  description: string;
  action: () => void;
}
```